src/main/java/App.java

```diff
@@ -715,6 +717,30 @@ private void renderShadowMap() {
 	private void loop() {
 		System.out.println("Render loop started");
 		while (!glfwWindowShouldClose(window)) {
+
+			// Generate reasonable random data collection positions if enabled
+			if (parameters.input.random.enabled) {
+				float min = -130.f / 2;
+				float max = 130.f / 2;
+				float rX = min + (float) Math.random() * (max - min);
+				float rZ = min + (float) Math.random() * (max - min);
+				float rYaw = (float) Math.random() * 359.f;
+				float rPitch = -30.f + (float) Math.random() * (30.f - (-30.f));
+				float rHeight = 0.15f + (float) Math.random() * (1.f - 0.3f);
+
+				float height = quadtree.getHeight(rX, rZ) + rHeight;
+
+				camera.setPosition(new Vector3f(rX, height, rZ));
+
+				camera.setYaw(rYaw);
+				camera.setPitch(rPitch);
+				camera.updateVectorsFromAngles();
+
+				if (frame >= parameters.input.random.endMax - 1) {
+					exit();
+				}
```

Comment:
Might be best to extract this into a separate method for readability but I'll leave that up to your discresion

Reply:
Thanks for the suggestion. I've extracted this into updateRandomPosition().